### ISSUE TITLE
Default to dark/light mode

### ISSUE THREAD

**Issue by shohamc1** [retypeapp/retype#421]
Right now it does not seem to be possible to set the default theme of the website and the theme depends on the user's browser preferences.

It would be great to add a configuration parameter that allows preferring a dark or light theme, and also completely disabling a theme (only have light mode).

**Comment by granicz** [comment]
Just a thought: I use dark mode by default in my browser, and there is nothing more annoying than coming across a website that decides to override that user preference, and instead presenting a light mode rendering. It just absolutely kills the eyes when everything else is dark.

**Comment by tjackman-ivre** [comment]
+1 to allow the option to disable dark mode. Depending on the usage, light looks more professional.

**Comment by latenitefilms** [comment]
I would also love to see this option - as dark mode actually better suits the visual style of our application. 

I would like to have dark mode by default.

**Comment by micartey** [comment]
Any news on this?

**Comment by tim-dreier** [comment]
Any news?

**Comment by latenitefilms** [comment]
I came across this thread again because I was looking for this feature - would love to see it!

**Comment by g1bbzy** [comment]
I came across this problem too. wanted to set the default theme to dark.
My solution.
add some custom js

- create a folder in the root directory called '_includes'
- inside that folder create a file called head.html
- inside that file add some javscript to set the default theme to dark or light

<script>
if (!window.localStorage.getItem('doc_theme')) {
	window.localStorage.setItem('doc_theme', 'dark');
	location.reload();
}
</script>

**Comment by geoffreymcgill** [comment]
I'm going to try and get this working for Retype `v3.9.0` and will post an update once complete.

**Comment by ferenc-a** [comment]
I'd also like this feature, preferably with a toggle between system / light / dark, with system (automatic) as the default.

**Comment by geoffreymcgill** [comment]
> I'd also like this feature, preferably with a toggle between system / light / dark, with system (automatic) as the default.

We're working on it. 👍 With some luck the functionality will be included in the v3.9.0 release.

**Comment by latenitefilms** [comment]
Here's a workaround in the meantime:

Put a `head.html` inside a folder `/docs/_includes/` which has:

```
<script>
    if (!window.localStorage.getItem('doc_theme')) {
        window.localStorage.setItem('doc_theme', 'dark');
        location.reload();
    }

</script>
```

**Comment by micartey** [comment]
> [...] in the meantime

This issue is 3 long years old 😆

**Comment by geoffreymcgill** [comment]
Defaulting to `dark` or `light` mode has been implemented and will be included with the upcoming v3.9.0 release. 

**NOTE**: This feature is a `Pro` only feature and a key is required to enable.

To configure Retype to render in `dark` mode, add the following setting to your project `retype.yml` file:

```yaml
scheme:
  mode: dark
```

You can also explicitly render in `light` mode as well. The following sample demonstrates defaulting to `light` mode:

```yaml
scheme:
  mode: light
```

The default value is `system`, which is the current functionality, and will automatically switch between `dark` and `light` modes as per the visitors system preference.

Testing the `scheme.mode` locally without a Retype Pro key is possible by adding the following setting to your `retype.yml`:

```yaml
start:
  pro: true
```

If the visitor has explicitly selected `dark` or `light` mode by clicking the `Dark|Light` button in the upper-right of the menu, their scheme choice will take preference over the `scheme.mode` project setting. If a visitor has chosen to view the website in `light` mode, Retype will render the website in `light` mode, even if `mode: dark` is configured in the `retype.yml` project settings. 

Hope this helps.

**Comment by micartey** [comment]
OMG, they finally closed this issue after 3 long years... Just to make it a Pro feature 🫠

**Comment by ferenc-a** [comment]
The shipped feature does not seem to be controllable from the user end, like I said in my earlier comment. For example, on the retype.com site I can only toggle light and dark, not back to automatic. Should I open another issue for this?

**Comment by geoffreymcgill** [comment]
> I can only toggle light and dark, not back to automatic. Should I open another issue for this?

Correct. For now. We need to build a three-way switch to be able to toggle back to `system` as the preference. 

That three-way switch will be built as a separate feature and will be included in a future release. It's a high priority feature for us to build, but at the moment I do not have an eta on when it will be available.

I have logged this conversation and will update this thread once the feature is available.

**Comment by geoffreymcgill** [comment]
A new three-way menu for selecting between `Light`, `Dark`, or `System` has been implemented. You can see it working on the [retype.com](https://retype.com/) website. 

![Image](https://github.com/user-attachments/assets/9377ebf3-be61-4de2-9b9e-fde78b5c6997)

The new functionality is available in the current `@next` release which can be installed locally using the following command:

```
npm install retypeapp@next --global
```

The functionality will be included in the upcoming `v3.11.0` release. 

Hope this helps.
